spec/unit/provider/package/rpm_spec.rb

```diff
@@ -52,6 +52,14 @@
       expect { @provider.run_action(:any) }.to raise_error(Chef::Exceptions::Package)
     end
 
+    it "should accept uri formed source" do
+      @new_resource.source "http://example.com/ImageMagick-c++-6.5.4.7-7.el6_5.x86_64.rpm"
+      allow(@provider).to receive(:popen4).and_return(@status)
```

Comment:
While you are in here, would you mind updating this provider to use `shell_out` instead of `popen4`?